### PR TITLE
Unify varchar limits

### LINKED ISSUE #35877: Varchar fields database vs core limits difference
### Describe the bug and add attachments

There are multiple fields that have either different length limits in their ObjectModel definitions and in database.
Also, there are many columns that have no limit in ObjectModel at all and just get cropped randomly with no error message.

Simpliest solution would be to edit in on ObjectModel side, so there is no need to do any upgrade scripts.

### Fields that need database change

| Class | Column | Object model limit | Database limit | What is wrong | How to fix | New size
|  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
| Meta | url_rewrite | 255 | 254 | Size difference | Raise DB (needs autoupgrade) | 255
| order\Order | reference | none | 9 | Limit missing | Raise both, 9 is too small (needs autoupgrade) | 255
| tax\TaxRulesGroup | name | 64 | 50 | Size difference | Raise DB (needs autoupgrade) | 64
| Mail | recipient | 255 | 126 | Size difference | Raise DB (needs autoupgrade) | 255
| Mail | subject | 255 | 254 | Size difference | Raise DB (needs autoupgrade) | 255
| shop\ShopUrl | domain | 255 | 150 | Size difference | Raise DB (needs autoupgrade) | 255
| shop\ShopUrl | domain_ssl | 255 | 150 | Size difference | Raise DB (needs autoupgrade) | 255
| FeatureFlag | description_wording | 191 | 512 | Size difference | Lower database limit, lower ORM limit | 191
| FeatureFlag | label_wording | 191 | 512 | Size difference | Lower database limit, lower ORM limit | 191

### Fields that need only PHP change

| Class | Column | Object model limit | Database limit | What is wrong | How to fix | New size
|  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
| Alias | alias | 255 | 191 | Size difference | Lower ObjectModel limit | 191
| Tab | icon | 64 | 32 | Size difference | Raise DB (needs autoupgrade) | 64
| CMSCategory | link_rewrite | 64 | 128 | Size difference | Raise ObjectModel | 128
| CMSCategory | name | 64 | 128 | Size difference | Raise ObjectModel | 128
| CustomerMessage | ip_address | 15 | 16 | Size difference | Raise ObjectModel | 16
| Profile | name | 32 | 128 | Size difference | Raise ObjectModel | 128
| Tab | name | 64 | 128 | Size difference | Raise ObjectModel | 128
| TranslatedConfiguration | name | 32 | 254 | Size difference | Raise ObjectModel | 254
| Address | vat_number | none | 32 | Limit missing | Add object model limit | -
| AddressFormat | format | none | 255 | Limit missing | Add object model limit | -
| AttributeGroup | group_type | none | 255 | Limit missing | Add object model limit | -
| Carrier | url | none | 255 | Limit missing | Add object model limit | -
| Connection | http_referer | none | 255 | Limit missing | Add object model limit | -
| ConnectionsSource | http_referer | none | 255 | Limit missing | Add object model limit | -
| ConnectionsSource | keywords | none | 255 | Limit missing | Add object model limit | -
| ConnectionsSource | request_uri | none | 255 | Limit missing | Add object model limit | -
| Country | zip_code_format | none | 12 | Limit missing | Add object model limit | -
| Customer | ape | none | 6 | Limit missing | Add object model limit | -
| Customer | company | none | 255 | Limit missing | Add object model limit | -
| Customer | ip_registration_newsletter | none | 15 | Limit missing | Add object model limit | -
| Customer | secure_key | none | 32 | Limit missing | Add object model limit | -
| Customer | siret | none | 14 | Limit missing | Add object model limit | -
| Customer | website | none | 128 | Limit missing | Add object model limit | -
| CustomerMessage | file_name | none | 18 | Limit missing | Add object model limit | -
| CustomerMessage | user_agent | none | 128 | Limit missing | Add object model limit | -
| CustomerThread | token | none | 12 | Limit missing | Add object model limit | -
| Hook | title | none | 255 | Limit missing | Add object model limit | -
| Manufacturer | meta_keywords | none | 255 | Limit missing | Add object model limit | -
| order\Order | module | none | 255 | Limit missing | Add object model limit | -
| order\Order | payment | none | 255 | Limit missing | Add object model limit | -
| order\Order | secure_key | none | 32 | Limit missing | Add object model limit | -
| order\OrderCarrier | tracking_number | none | 64 | Limit missing | Add object model limit | -
| order\OrderCartRule | name | none | 254 | Limit missing | Add object model limit | -
| order\OrderDetail | download_hash | none | 255 | Limit missing | Add object model limit | -
| order\OrderDetail | product_ean13 | none | 13 | Limit missing | Add object model limit | -
| order\OrderDetail | product_isbn | none | 32 | Limit missing | Add object model limit | -
| order\OrderDetail | product_mpn | none | 40 | Limit missing | Add object model limit | -
| order\OrderDetail | product_reference | none | 64 | Limit missing | Add object model limit | -
| order\OrderDetail | product_supplier_reference | none | 64 | Limit missing | Add object model limit | -
| order\OrderDetail | product_upc | none | 12 | Limit missing | Add object model limit | -
| order\OrderDetail | tax_name | none | 16 | Limit missing | Add object model limit | -
| order\OrderPayment | payment_method | none | 255 | Limit missing | Add object model limit | -
| order\OrderReturnState | color | none | 32 | Limit missing | Add object model limit | -
| order\OrderState | color | none | 32 | Limit missing | Add object model limit | -
| order\OrderState | module_name | none | 255 | Limit missing | Add object model limit | -
| PrestaShopLogger | object_type | none | 32 | Limit missing | Add object model limit | -
| Product | unity | none | 255 | Limit missing | Add object model limit | -
| ProductAttribute | color | none | 32 | Limit missing | Add object model limit | -
| SearchEngine | getvar | none | 16 | Limit missing | Add object model limit | -
| SearchEngine | server | none | 64 | Limit missing | Add object model limit | -
| shop\Shop | color | none | 50 | Limit missing | Add object model limit | -
| shop\Shop | theme_name | none | 255 | Limit missing | Add object model limit | -
| shop\ShopGroup | color | none | 50 | Limit missing | Add object model limit | -
| SpecificPriceRule | name | none | 255 | Limit missing | Add object model limit | -
| stock\Stock | ean13 | none | 13 | Limit missing | Add object model limit | -
| stock\Stock | isbn | none | 32 | Limit missing | Add object model limit | -
| stock\Stock | mpn | none | 40 | Limit missing | Add object model limit | -
| stock\Stock | reference | none | 64 | Limit missing | Add object model limit | -
| stock\Stock | upc | none | 12 | Limit missing | Add object model limit | -
| stock\StockMvt | employee_firstname | none | 255 | Limit missing | Add object model limit | -
| stock\StockMvt | employee_lastname | none | 255 | Limit missing | Add object model limit | -
| stock\StockMvtWS | employee_firstname | none | 255 | Limit missing | Add object model limit | -
| stock\StockMvtWS | employee_lastname | none | 255 | Limit missing | Add object model limit | -
| tax\TaxRule | description | none | 100 | Limit missing | Add object model limit | -
| tax\TaxRule | zipcode_from | none | 12 | Limit missing | Add object model limit | -
| tax\TaxRule | zipcode_to | none | 12 | Limit missing | Add object model limit | -

### Fields OK

| Class | Column | Object model limit | Database limit | What is wrong | How to fix | New size
|  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
| Address | address1 | 128 | 128 | ✅ OK | - | -
| Address | address2 | 128 | 128 | ✅ OK | - | -
| Address | alias | 32 | 32 | ✅ OK | - | -
| Address | city | 64 | 64 | ✅ OK | - | -
| Address | company | 255 | 255 | ✅ OK | - | -
| Address | dni | 16 | 16 | ✅ OK | - | -
| Address | firstname | 255 | 255 | ✅ OK | - | -
| Address | lastname | 255 | 255 | ✅ OK | - | -
| Address | phone | 32 | 32 | ✅ OK | - | -
| Address | phone_mobile | 32 | 32 | ✅ OK | - | -
| Address | postcode | 12 | 12 | ✅ OK | - | -
| Alias | search | 255 | 255 | ✅ OK | - | -
| Attachment | file | 40 | 40 | ✅ OK | - | -
| Attachment | file_name | 128 | 128 | ✅ OK | - | -
| Attachment | mime | 128 | 128 | ✅ OK | - | -
| Attachment | name | 32 | 32 | ✅ OK | - | -
| AttributeGroup | name | 128 | 128 | ✅ OK | - | -
| AttributeGroup | public_name | 64 | 64 | ✅ OK | - | -
| Carrier | delay | 512 | 512 | ✅ OK | - | -
| Carrier | external_module_name | 64 | 64 | ✅ OK | - | -
| Carrier | name | 64 | 64 | ✅ OK | - | -
| Cart | secure_key | 32 | 32 | ✅ OK | - | -
| CartRule | code | 254 | 254 | ✅ OK | - | -
| CartRule | name | 254 | 254 | ✅ OK | - | -
| Category | link_rewrite | 128 | 128 | ✅ OK | - | -
| Category | meta_description | 512 | 512 | ✅ OK | - | -
| Category | meta_keywords | 255 | 255 | ✅ OK | - | -
| Category | meta_title | 255 | 255 | ✅ OK | - | -
| Category | name | 128 | 128 | ✅ OK | - | -
| CMS | head_seo_title | 255 | 255 | ✅ OK | - | -
| CMS | link_rewrite | 128 | 128 | ✅ OK | - | -
| CMS | meta_description | 512 | 512 | ✅ OK | - | -
| CMS | meta_keywords | 255 | 255 | ✅ OK | - | -
| CMS | meta_title | 255 | 255 | ✅ OK | - | -
| CMSCategory | meta_description | 512 | 512 | ✅ OK | - | -
| CMSCategory | meta_keywords | 255 | 255 | ✅ OK | - | -
| CMSCategory | meta_title | 255 | 255 | ✅ OK | - | -
| CMSRole | name | 50 | 50 | ✅ OK | - | -
| Combination | available_later | 255 | 255 | ✅ OK | - | -
| Combination | available_now | 255 | 255 | ✅ OK | - | -
| Combination | ean13 | 13 | 13 | ✅ OK | - | -
| Combination | isbn | 32 | 32 | ✅ OK | - | -
| Combination | mpn | 40 | 40 | ✅ OK | - | -
| Combination | reference | 64 | 64 | ✅ OK | - | -
| Combination | supplier_reference | 64 | 64 | ✅ OK | - | -
| Combination | upc | 12 | 12 | ✅ OK | - | -
| Configuration | name | 254 | 254 | ✅ OK | - | -
| Contact | email | 255 | 255 | ✅ OK | - | -
| Contact | name | 255 | 255 | ✅ OK | - | -
| Country | iso_code | 3 | 3 | ✅ OK | - | -
| Country | name | 64 | 64 | ✅ OK | - | -
| Currency | iso_code | 3 | 3 | ✅ OK | - | -
| Currency | name | 255 | 255 | ✅ OK | - | -
| Currency | numeric_iso_code | 3 | 3 | ✅ OK | - | -
| Currency | pattern | 255 | 255 | ✅ OK | - | -
| Currency | symbol | 255 | 255 | ✅ OK | - | -
| Customer | email | 255 | 255 | ✅ OK | - | -
| Customer | firstname | 255 | 255 | ✅ OK | - | -
| Customer | lastname | 255 | 255 | ✅ OK | - | -
| Customer | passwd | 255 | 255 | ✅ OK | - | -
| Customer | reset_password_token | 40 | 40 | ✅ OK | - | -
| CustomerSession | token | 40 | 40 | ✅ OK | - | -
| CustomerThread | email | 255 | 255 | ✅ OK | - | -
| CustomizationField | name | 255 | 255 | ✅ OK | - | -
| Employee | bo_color | 32 | 32 | ✅ OK | - | -
| Employee | bo_css | 64 | 64 | ✅ OK | - | -
| Employee | bo_theme | 32 | 32 | ✅ OK | - | -
| Employee | email | 255 | 255 | ✅ OK | - | -
| Employee | firstname | 255 | 255 | ✅ OK | - | -
| Employee | lastname | 255 | 255 | ✅ OK | - | -
| Employee | passwd | 255 | 255 | ✅ OK | - | -
| Employee | preselect_date_range | 32 | 32 | ✅ OK | - | -
| Employee | reset_password_token | 40 | 40 | ✅ OK | - | -
| EmployeeSession | token | 40 | 40 | ✅ OK | - | -
| Feature | name | 128 | 128 | ✅ OK | - | -
| FeatureFlag | description_domain | 255 | 255 | ✅ OK | - | -
| FeatureFlag | label_domain | 255 | 255 | ✅ OK | - | -
| FeatureFlag | name | 191 | 191 | ✅ OK | - | -
| FeatureFlag | stability | 64 | 64 | ✅ OK | - | -
| FeatureValue | value | 255 | 255 | ✅ OK | - | -
| Gender | name | 20 | 20 | ✅ OK | - | -
| Group | name | 32 | 32 | ✅ OK | - | -
| Guest | accept_language | 8 | 8 | ✅ OK | - | -
| Hook | name | 191 | 191 | ✅ OK | - | -
| Image | legend | 128 | 128 | ✅ OK | - | -
| ImageType | name | 64 | 64 | ✅ OK | - | -
| Language | date_format_full | 32 | 32 | ✅ OK | - | -
| Language | date_format_lite | 32 | 32 | ✅ OK | - | -
| Language | iso_code | 2 | 2 | ✅ OK | - | -
| Language | language_code | 5 | 5 | ✅ OK | - | -
| Language | locale | 5 | 5 | ✅ OK | - | -
| Language | name | 32 | 32 | ✅ OK | - | -
| Mail | template | 62 | 62 | ✅ OK | - | -
| Manufacturer | meta_description | 512 | 512 | ✅ OK | - | -
| Manufacturer | meta_title | 255 | 255 | ✅ OK | - | -
| Manufacturer | name | 64 | 64 | ✅ OK | - | -
| Meta | description | 255 | 255 | ✅ OK | - | -
| Meta | keywords | 255 | 255 | ✅ OK | - | -
| Meta | page | 64 | 64 | ✅ OK | - | -
| Meta | title | 128 | 128 | ✅ OK | - | -
| order\OrderMessage | name | 128 | 128 | ✅ OK | - | -
| order\OrderPayment | card_brand | 254 | 254 | ✅ OK | - | -
| order\OrderPayment | card_expiration | 254 | 254 | ✅ OK | - | -
| order\OrderPayment | card_holder | 254 | 254 | ✅ OK | - | -
| order\OrderPayment | card_number | 254 | 254 | ✅ OK | - | -
| order\OrderPayment | order_reference | 9 | 9 | ✅ OK | - | -
| order\OrderPayment | transaction_id | 254 | 254 | ✅ OK | - | -
| order\OrderReturnState | name | 64 | 64 | ✅ OK | - | -
| order\OrderState | name | 64 | 64 | ✅ OK | - | -
| order\OrderState | template | 64 | 64 | ✅ OK | - | -
| Product | available_later | 255 | 255 | ✅ OK | - | -
| Product | available_now | 255 | 255 | ✅ OK | - | -
| Product | delivery_in_stock | 255 | 255 | ✅ OK | - | -
| Product | delivery_out_stock | 255 | 255 | ✅ OK | - | -
| Product | ean13 | 13 | 13 | ✅ OK | - | -
| Product | isbn | 32 | 32 | ✅ OK | - | -
| Product | link_rewrite | 128 | 128 | ✅ OK | - | -
| Product | location | 255 | 255 | ✅ OK | - | -
| Product | meta_description | 512 | 512 | ✅ OK | - | -
| Product | meta_keywords | 255 | 255 | ✅ OK | - | -
| Product | meta_title | 255 | 255 | ✅ OK | - | -
| Product | mpn | 40 | 40 | ✅ OK | - | -
| Product | name | 128 | 128 | ✅ OK | - | -
| Product | reference | 64 | 64 | ✅ OK | - | -
| Product | supplier_reference | 64 | 64 | ✅ OK | - | -
| Product | upc | 12 | 12 | ✅ OK | - | -
| ProductAttribute | name | 128 | 128 | ✅ OK | - | -
| ProductDownload | display_filename | 255 | 255 | ✅ OK | - | -
| ProductDownload | filename | 255 | 255 | ✅ OK | - | -
| ProductSupplier | product_supplier_reference | 64 | 64 | ✅ OK | - | -
| QuickAccess | link | 255 | 255 | ✅ OK | - | -
| QuickAccess | name | 32 | 32 | ✅ OK | - | -
| RequestSql | name | 200 | 200 | ✅ OK | - | -
| Risk | color | 32 | 32 | ✅ OK | - | -
| Risk | name | 20 | 20 | ✅ OK | - | -
| shop\Shop | name | 64 | 64 | ✅ OK | - | -
| shop\ShopGroup | name | 64 | 64 | ✅ OK | - | -
| shop\ShopUrl | physical_uri | 64 | 64 | ✅ OK | - | -
| shop\ShopUrl | virtual_uri | 64 | 64 | ✅ OK | - | -
| State | iso_code | 7 | 7 | ✅ OK | - | -
| State | name | 80 | 80 | ✅ OK | - | -
| stock\StockAvailable | location | 255 | 255 | ✅ OK | - | -
| stock\StockMvtReason | name | 255 | 255 | ✅ OK | - | -
| Store | address1 | 255 | 255 | ✅ OK | - | -
| Store | address2 | 255 | 255 | ✅ OK | - | -
| Store | city | 64 | 64 | ✅ OK | - | -
| Store | email | 255 | 255 | ✅ OK | - | -
| Store | fax | 16 | 16 | ✅ OK | - | -
| Store | name | 255 | 255 | ✅ OK | - | -
| Store | phone | 16 | 16 | ✅ OK | - | -
| Store | postcode | 12 | 12 | ✅ OK | - | -
| Supplier | meta_description | 512 | 512 | ✅ OK | - | -
| Supplier | meta_keywords | 255 | 255 | ✅ OK | - | -
| Supplier | meta_title | 255 | 255 | ✅ OK | - | -
| Supplier | name | 64 | 64 | ✅ OK | - | -
| Tab | class_name | 64 | 64 | ✅ OK | - | -
| Tab | module | 64 | 64 | ✅ OK | - | -
| Tab | route_name | 256 | 256 | ✅ OK | - | -
| Tab | wording | 255 | 255 | ✅ OK | - | -
| Tab | wording_domain | 255 | 255 | ✅ OK | - | -
| Tag | name | 32 | 32 | ✅ OK | - | -
| tax\Tax | name | 32 | 32 | ✅ OK | - | -
| webservice\WebserviceKey | key | 32 | 32 | ✅ OK | - | -
| Zone | name | 64 | 64 | ✅ OK | - | -

### Expected behavior

All limits must be unified.

### PrestaShop version(s) where the bug happened

8.1.5

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Unifies VARCHAR/STRING field limits in object model/ORM vs the database. See the commit list for each change.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Tests green. Manually check that every db_structure.sql change matches object model opposite lengths in my PR. Manually check that every object model change matches db_structure.sql opposite lengths in my PR.
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/10598665633
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/35877
| Related PRs       | https://github.com/PrestaShop/autoupgrade/pull/869
| Sponsor company   |

### CODE REVIEW

File: install-dev/data/db_structure.sql [hunk 1]
```diff
@@ -47,7 +47,7 @@ CREATE TABLE `PREFIX_address` (
 /* Used for search, if a search string is present inside the table, search the alias as well */
 CREATE TABLE `PREFIX_alias` (
   `id_alias` int(10) unsigned NOT NULL auto_increment,
-  `alias` varchar(191) NOT NULL,
+  `alias` varchar(255) NOT NULL,
```
Comment (kpodemski) [inline install-dev/data/db_structure.sql]: This should not be changed and rather fixed on the ObjectModel level, see: https://github.com/PrestaShop/PrestaShop/commit/61b47cce5b701bf92d091339c359e49ce3b2fe10

This is only for those fields which are indexes.

This PR should go to the develop branch to make sure we won't mistakenly introduce problems in a patch version.

**Review by kpodemski — DISMISSED** [review]
This PR should go to the develop branch, see my comment.

File: install-dev/data/db_structure.sql (hunk 1 above)
Comment (jolelievre) [reply install-dev/data/db_structure.sql]: Indeed you should set 191 as the limit in Address::alias instead

**Review by marsaldev — DISMISSED** [review]
Fix AddressCore - row 132
- alias size to 191

Revert change in db_structure.sql - row 50
@Hlavtox

File: install-dev/data/db_structure.sql (hunk 1 above)
Comment (Hlavtox) [reply install-dev/data/db_structure.sql]: Resolved

**Comment by Hlavtox** [comment]
@marsaldev @kpodemski @jolelievre Ready

**Comment by Hlavtox** [comment]
@kpodemski @jolelievre @PrestaShop/committers Can someone review please? :-)

**Comment by Hlavtox** [comment]
@kpodemski @jolelievre @PrestaShop/committers Hi again, can someone review please? :-)

**Review by FabienPapet — APPROVED** [review]

**Review by ShaiMagal — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Hlavtox ,

Checked all the data mentioned in linked issue.

Fields that need database change -> checked in DB ✅ 
Fields that need only PHP change -> checed in PHP files✅ 

Thanks for the PR 💪 
It is QA ✅

**Comment by nicosomb** [comment]
@Hlavtox could you please open a PR on autoupgrade project?

**Comment by Hlavtox** [comment]
@nicosomb Done - https://github.com/PrestaShop/autoupgrade/pull/869

**Pull request merged**